pkg/controller/storagecluster/cephcluster.go

```diff
@@ -321,13 +330,34 @@ func getMonCount(nodeCount int) int {
 	return count
 }
 
+// determineKubeVersion gets the kube server version
+func determineKubeVersion() (*version.Info, error) {
+	clientset, err := kubernetes.NewForConfig(config.GetConfigOrDie())
+	if err != nil {
+		return nil, err
+	}
+
+	serverVersion, err := clientset.Discovery().ServerVersion()
+	if err != nil {
+		return nil, err
+	}
+
+	return serverVersion, nil
+}
+
 // newStorageClassDeviceSets converts a list of StorageDeviceSets into a list of Rook StorageClassDeviceSets
-func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rook.StorageClassDeviceSet {
+func newStorageClassDeviceSets(sc *ocsv1.StorageCluster, serverVersion *version.Info) []rook.StorageClassDeviceSet {
 	storageDeviceSets := sc.Spec.StorageDeviceSets
 	topologyMap := sc.Status.NodeTopologies
 
+	var noPlacement bool
+	var noPreparePlacement bool
```

Comment:
I do see the need for defining them before instnatiating them, but why are you defining them up here? I can't see anything that would mean we need them outside the scope of the `for` loop they were originally in, but I might just be blind.

Reply:
fixed!